### ISSUE TITLE
[20.03] Nautilus unable to open GVFS paths

### ISSUE THREAD

**Issue by onny** [NixOS/nixpkgs#89542]
**Describe the bug**
In Nautilus I'm unable to browse special paths like mtp, smb, sftp etc. None of these protocols seems to be available.
According to the documentation, I just have to add
```
services.gvfs.enable = true;
```
But this doesn't seem to work. I'm using Sway as a window manager, started directly without a display manager.

**To Reproduce**
1. Install ``nautilus`` and enable ``gvfs``
2. Run ``nautilus``, click on ``Other locations`` and try to open for example ``ssh://192.168.178.2``
3. Unable to open, no protocols are available

**Expected behavior**
Nautilus should be able to open gvfs volumes.

**Additional context**
My complete NixOS system configuration https://git.project-insanity.org/onny/nixos-lappy

**Notify maintainers**
@worldofpeace @lethalman @jtojnar @hedning  

**Metadata**
```
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/hzrq5ccamfkwdkl9gqqsh3ryprzvs044-bash-interactive-4.4-p23-dev
copying path '/nix/store/hzrq5ccamfkwdkl9gqqsh3ryprzvs044-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.15, NixOS, 20.03.2059.05a32d8e771 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.5`
 - channels(root): `"nixos-20.03.2059.05a32d8e771"`
 - channels(onny): `"nixos-20.03.1422.1e90c46c2d9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by onny** [comment]
Still no protocols available :(

![2020-07-13-124405_grim](https://user-images.githubusercontent.com/757752/87296158-a19b3300-c506-11ea-9c55-8fecc68c2403.png)

@hedning @jtojnar @worldofpeace @dasj19

**Comment by dasj19** [comment]
Is this 20.03 or unstable? I use unstable and I just made use of smb:// yesterday... all the other protocols were present also.

**Comment by onny** [comment]
> Is this 20.03 or unstable? I use unstable and I just made use of smb:// yesterday... all the other protocols were present also.

Thanks for the response. Here's the version information
```
onny@tuxzentrale ~> nix-channel --list
nixos https://nixos.org/channels/nixos-20.03

onny@tuxzentrale ~> nautilus --version

** (org.gnome.Nautilus:24448): WARNING **: 12:55:14.895: Error on getting connection: Failed to load SPARQL backend: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Tracker1 was not provided by any .service files

(org.gnome.Nautilus:24448): dbind-WARNING **: 12:55:14.930: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
GNOME nautilus 3.34.2
onny@tuxzentrale ~> 
```

**Comment by dasj19** [comment]
Thanks for providing additional info. I personally run on unstable and I am new in the gnome team.
But first things first.
1. Prefix this PR with [20.03]
2. There is a new release of nautilus on gnome 3.34 branch: https://download.gnome.org/sources/nautilus/3.34/
Do you know how to make a package update?

**Comment by onny** [comment]
Okay thanks! Not sure how to update, should I try the unstable channel?

Some more information about the GIO_EXTRA_MODULES variable:
```
<code>
onny@tuxzentrale ~> echo $GIO_EXTRA_MODULES
/nix/store/xy4bmni2icmrcjdrxjyxvv5msl7djpzv-dconf-0.34.0-lib/lib/gio/modules:/nix/store/xy4bmni2icmrcjdrxjyxvv5msl7djpzv-dconf-0.34.0-lib/lib/gio/modules

onny@tuxzentrale ~> ls -lahgs /nix/store/xy4bmni2icmrcjdrxjyxvv5msl7djpzv-dconf-0.34.0-lib/lib/gio/modules
total 84K
   0 dr-xr-xr-x 1 root  68 Jan  1  1970 ./
   0 dr-xr-xr-x 1 root  14 Jan  1  1970 ../
4.0K -r--r--r-- 1 root  39 Jan  1  1970 giomodule.cache
 80K -r-xr-xr-x 1 root 80K Jan  1  1970 libdconfsettings.so*
```

**Comment by dasj19** [comment]
No need to try unstable yet... I'll jump on 20.03 branch and make an update for you.

**Comment by dasj19** [comment]
So I opened a PR for you. https://github.com/NixOS/nixpkgs/pull/93056

Can you help with testing?

Clone this repository in a folder of your choice. Then Run the automatic reviews:
```
git@github.com:NixOS/nixpkgs.git
git checkout 20.03
nix-shell -p nixpkgs-review --run "nixpkgs-review pr 93056"
```
And see if it builds. Thanks

**Comment by onny** [comment]
Wow nice, this looks much more better now :)

![2020-07-13-175957_grim](https://user-images.githubusercontent.com/757752/87326199-b988ac00-c532-11ea-914d-aeb6d37e04d9.png)

```
[nix-shell:~/.cache/nixpkgs-review/pr-93056-1]$ pwd
/home/onny/.cache/nixpkgs-review/pr-93056-1

[nix-shell:~/.cache/nixpkgs-review/pr-93056-1]$ nautilus --version
GNOME nautilus 3.34.3
```

Hope this get merged quite soon :)

**Comment by dasj19** [comment]
As discussion moved here: https://github.com/NixOS/nixpkgs/pull/93056
The gnome guys have a huge gnome-34 update waiting around the corner so that will make my pull request obsolete. (Because they are updating nautilus as well)

**Comment by jtojnar** [comment]
I would not expect Nautilus update to fix this, see the [changes](https://ftp.gnome.org/pub/GNOME/sources/nautilus/3.34/nautilus-3.34.3.news) for that release:

> Major changes in 3.34.3:
> * Revert icon emblem fixes in order to prevent performance issues (António Fernandes)
> * Fix crashes often happening when searching (Ondrej Holy)
> * Fix crashes after conflict dialog response (Ondrej Holy)

The reason it works is likely that gvfs module is added to `GIO_EXTRA_MODULES` by the shell but not by the wrapper. We need to add it to the wrapper.

**Comment by jtojnar** [comment]
Just adding `gvfs` to nautilus’s `buildInputs` should fix it.

**Comment by onny** [comment]
> I would not expect Nautilus update to fix this, see the [changes](https://ftp.gnome.org/pub/GNOME/sources/nautilus/3.34/nautilus-3.34.3.news) for that release:
> 
> > Major changes in 3.34.3:
> > 
> > * Revert icon emblem fixes in order to prevent performance issues (António Fernandes)
> > * Fix crashes often happening when searching (Ondrej Holy)
> > * Fix crashes after conflict dialog response (Ondrej Holy)
> 
> The reason it works is likely that gvfs module is added to `GIO_EXTRA_MODULES` by the shell but not by the wrapper. We need to add it to the wrapper.

I can confirm this that when I use the contents of the GIO_EXTRA_MODULES variable used by the shell, Nautilus seems to work well even with the older version:

```
onny@tuxzentrale ~> nautilus --version

(org.gnome.Nautilus:13120): dbind-WARNING **: 20:20:15.138: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
GNOME nautilus 3.34.2

onny@tuxzentrale ~> env GIO_EXTRA_MODULES="/nix/store/7fnxbqf0mwj64549h4z2d9zbxsmb3ip6-gvfs-1.42.2/lib/gio/modules:/nix/store/xy4bmni2icmrcjdrxjyxvv5msl7djpzv-dconf-0.34.0-lib/lib/gio/modules" nautilus
``` 
:+1: 

So hopefully gvfs gio modules would be added to ``GIO_EXTRA_MODULES`` as soon ``gvfs`` is part of Nautilus ``buildInputs`` :)

**Comment by worldofpeace** [comment]
> As discussion moved here: #93056
> The gnome guys have a huge gnome-34 update waiting around the corner so that will make my pull request obsolete. (Because they are updating nautilus as well)
> This PR will take over: #85974

@das-g Yeah, we really should finish that gnome-34 update for 20.03. I believe that PR is entirely obsolete now because it wasn't merged. The main issue is it needed some build Q.A, and the fact that we don't have any automatic testing to just run the main applications makes it rather difficult. Perhaps we can setup a jobset gnome-20.03 so we can test updates for each release clearly.

**Comment by das-g** [comment]
> @das-g Yeah, we really should finish that gnome-34 update for 20.03.

Did you mean to mention @dasj19 rather than me, @worldofpeace?

**Comment by worldofpeace** [comment]
> > @das-g Yeah, we really should finish that gnome-34 update for 20.03.
> 
> Did you mean to mention @dasj19 rather than me, @worldofpeace?

I did, sorry for the mention. Your @'s are very close, especially when it comes to autosuggestion :grinning:

**Comment by flokli** [comment]
Successfully mounted a `davs://` share from nautilus on 20.03 with https://github.com/NixOS/nixpkgs/pull/93056 merged, so this can be closed.

**Comment by onny** [comment]
Issue is still present for me. I'm using ``nixos-unstable`` with ``nautilus`` 3.36.3 and all protocols are missing :(

**Comment by flokli** [comment]
@onny can you try again by running `$(nix-build nixos/tests/gnome3.nix -A driver)/bin/nixos-run-vms` from a checkout of latest master?

I tried with both `master` and the `release-20.03` branch. Opened Nautilus, went to "Other Locations".

The question mark shows `davs://` URLs (and others), and I could successfully connect and list files from there.

**Comment by con-f-use** [comment]
I'm having the same problem with 3.36.3 on NixOS 20.09.

**Comment by bjornfor** [comment]
@con-f-use: Weird. It works for me on NixOS 20.09. Either we need to dig into config specifics, or there is some state somewhere that breaks it. (FYI, I think this issue was fixed for me on nixos-20.03 just by upgrading, and just now I upgraded to nixos-20.09 and it still works. I didn't reset or wipe any state that I remember.)

**Comment by con-f-use** [comment]
I noticed, that I also can't change preferences in nautilus: [zipped vide](https://github.com/NixOS/nixpkgs/files/5581440/nautilus_settings.zip) (because github).

Might be related to me just using nautilus without gnome as a DE and/or this:
```
$ gsettings list-schemas
No schemas installed
```

There might be something missing, that's actually a dependency of nautilus.
